protocol/network/7-properties.md

```diff
@@ -1,16 +1,105 @@
+7 Properties
+============
 
-# IPFS Protocol on the Wire
+## 7.1 Communication Model - Streams
 
-Authors: [Juan Benet](http://github.com/jbenet)
+The Network layer handles all the problems of connecting to a peer, and exposes
+simple bidirectional streams. Users can both open a new stream
+(`NewStream()`) and register a stream handler (`SetStreamHandler`). The user
+is then free to implement whatever wire messaging protocol she desires. This
+makes it easy to build peer-to-peer protocols, as the complexities of
+connectivity, multi-transport support, flow control, and so on, are handled.
```

Comment:
we should probably choose between:
```
- NewStream() or OpenStream()
- SetHandler() or SetStreamHandler()
```
if we want to emulate the plan9 socket interface, we can do:

```
DialStream(protocol.ID)
ListenStream(protocol.ID, HandlerFunc)
```